p2p/peer.go

```diff
@@ -317,11 +329,47 @@ func (p *peer) NodeInfo() *NodeInfo {
 	return &n
 }
 
+// MetricStore - Returns the peer's TrustMetricStore
+func (p *peer) MetricStore() *trust.TrustMetricStore {
+	return p.metricStore
+}
+
+// SetMetricStore - Sets the switch's TrustMetricStore for allowing all reactors to access the metrics
+func (p *peer) SetMetricStore(tms *trust.TrustMetricStore) {
+	p.metricStore = tms
+}
+
 // Status returns the peer's ConnectionStatus.
 func (p *peer) Status() ConnectionStatus {
 	return p.mconn.Status()
 }
 
+// It updates a peer's metric with good or bad events taking place within this reactor
+func (p *peer) Mark(reactorID string, good bool, events, severity int) {
```

Comment:
I'd create 2 funcs

```
MarkAsGoodNEvents(reactorID string, events int, s severity)
MarkAsBadNEvents(reactorID string, events int, s severity)
```

src.Mark(conR.GetID(), false, 1, p2p.PeerMarkBad) is not the best API because you need to look docs or method signature in order to figure out what false and 1 stands for.

```
src.Mark(memR.GetID(), true, 1, p2p.PeerMarkCorrect)
```

versus

```
src.MarkAsGoodNEvents(memR.GetID(), 1, p2p.PeerMarkCorrect)
```

Reply:
Other question is: should peers be aware they are part of the reactors. i.e. maybe this `reactorID string` should be really `peerTrustMetricValue string`?

Reply:
Okay, I have created the two separate methods, and renamed the parameter to trustMetricID.